x-pack/plugins/apm/public/components/shared/transactions_table/index.tsx

```diff
@@ -77,24 +85,37 @@ export function TransactionsTable({
   kuery,
   start,
   end,
+  saveTableOptionsToUrl = false,
 }: Props) {
-  const [tableOptions] = useState<{
-    pageIndex: number;
-    sort: {
-      direction: SortDirection;
-      field: SortField;
-    };
+  const history = useHistory();
+
+  const {
+    urlParams: {
+      page: urlPage = 0,
+      pageSize: urlPageSize = numberOfTransactionsPerPage,
+      sortField: urlSortField = 'impact',
+      sortDirection: urlSortDirection = 'desc',
+    },
+  } = useLegacyUrlParams();
```

Comment:
Why we need to use the `useLegacyUrlParams()` here? Can we use `useApmParams`?

Reply:
I'd say that it's because this component is used by more than um page (path), so if I do `useApmParams('/*')` page is not defined there.

Reply:
Thanks for that. There you are. https://github.com/elastic/kibana/pull/128287/commits/43965eb7e527303815536cab5c408d4c771bb29e